### PR TITLE
[P2T support No.9、10、48、49、 84、 90、 102、 107、 111][BIT] nn.functional geometric weight_quantize clip reshape unflatten

### PR DESCRIPTION
**Pull request by cszdrg**
# 直接映射
- paddle.poisson **pass**
# 有对应api 但需要rule
- paddle.reshape **pass**
- paddle.Tensor.clip **pass**
- paddle.unflatten **pass**
- paddle.clip **accuracy error** 存在较大误差
需要规范最大最小值
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.clip.html
# 无对应api 需要rule
- paddle.geometric.send_u_recv  **pass**
- paddle.geometric.send_ue_recv **pass**
- paddle.nn.functional.label_smooth **pass**
- paddle.nn.functional.log_loss **pass**
- paddle.nn.functional.margin_cross_entropy **pass**
- paddle.nn.functional.maxout **pass**
- paddle.nn.functional.npair_loss **pass**
- paddle.nn.functional.rnnt_loss **pass**
- paddle.nn.functional.sequence_mask **pass**
- paddle.nn.functional.square_error_cost **pass**
- paddle.nn.functional.swish **pass**
- paddle.nn.functional.temporal_shift **pass**
- paddle.nn.functional.zeropad2d **pass**
- paddle.nn.functional.softmax_with_cross_entropy **acuracy** 存在大误差
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/softmax_with_cross_entropy_cn.html
- paddle.nn.functional.sigmoid_focal_loss **acuracy**  存在大误差
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/sigmoid_focal_loss_cn.html
- paddle.nn.functional.dice_loss  **acuracy** 小误差0.07
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/dice_loss_cn.html
- paddle.nn.quant.weight_only_linear
**误差较大 放入not support 不支持**
- paddle.nn.quant.weight_quantize
**误差较大 放入not support 不支持**

### CODE REVIEW

File: tester/paddle_to_torch/rules.py [hunk 1]
```diff
@@ -2451,7 +2483,36 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+
+batch_size = input.size(0)
+if num_classes is None:
+    num_classes = input.size(1)
+
+# 获取每个样本的目标 logit 值
+target_logits = input[torch.arange(batch_size), label]
+
+print(target_logits.shape)
+
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2500]: 删去print

File: tester/paddle_to_torch/rules.py [hunk 2]
```diff
@@ -2451,7 +2483,36 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+if bias is not None:
+    target_logits += bias[label]
+
+# BCE with logits: -log( sigmoid(target_logit) )
+loss = torch.nn.functional.binary_cross_entropy_with_logits(target_logits, torch.ones_like(target_logits), reduction='none')
 
+if weights is not None:
+    loss = loss * weights
+result = loss
+print(result)
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:2510]: 同上

File: tester/paddle_to_torch/rules.py [hunk 3]
```diff
@@ -2890,6 +3017,22 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+        core = """
+axis = locals().get('axis', 1)
+axis = axis if axis >= 0 else x.dim() + axis
+in_channels = x.shape[axis]
+channels_per_group = in_channels // groups
+shape = list(x.shape)
+new_shape = shape[:axis] + [channels_per_group, groups] + shape[axis+1:]
+    
+x = x.reshape(*new_shape)
+print(x.shape)
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:3031]: 同上

File: tester/paddle_to_torch/rules.py [hunk 4]
```diff
@@ -3971,7 +4147,25 @@ def apply(self, paddle_api: str) -> ConvertResult:
         )
         return ConvertResult.success(paddle_api, code)
 
-
+class RnntLossRule(BaseRule):
+    def apply(self, paddle_api: str) -> ConvertResult:
+        core = """
+import torchaudio
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:4153]: 这个是否会报错？

File: tester/paddle_to_torch/rules.py [hunk 5]
```diff
@@ -4931,6 +5436,20 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+data_format = locals().get('data_format', 'NCHW')
+pad_left, pad_right, pad_top, pad_bottom = padding
+if data_format == "NHWC":
+    x = x.permute(0, 3, 1, 2)
+    padded = torch.nn.functional.pad(x, (pad_left, pad_right, pad_top, pad_bottom), mode="constant", value=0)
+    result = padded.permute(0, 2, 3, 1)
+elif data_format == "NCHW":
+    result = torch.nn.functional.pad(x, (pad_left, pad_right, pad_top, pad_bottom), mode="constant", value=0)
+"""
+        code = impl.splitlines()
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/rules.py:5451]: 使用Code数据类

**Review by cangtianhuang — CHANGES_REQUESTED** [review]
考虑 paddle.nn.functional.softmax_with_cross_entropy、paddle.nn.functional.sigmoid_focal_loss 是否是实现有误😉

File: tester/api_config/api_config_notsupport_paddle2torch_4.txt
Comment (yuwu46) [inline tester/api_config/api_config_notsupport_paddle2torch_4.txt:1]: 请把文件名改为api_config_paddleonly_4.txt

File: tester/paddle_to_torch/rules.py (hunk 5 above)
Comment (cszdrg) [reply tester/paddle_to_torch/rules.py:5451]: 好的

File: tester/paddle_to_torch/rules.py (hunk 4 above)
Comment (cszdrg) [reply tester/paddle_to_torch/rules.py:4153]: 不会报错
